src/utils/issue.ts

```diff
@@ -19,36 +20,27 @@ export interface IssueResult {
 
 export async function issue(
     api: ApiPromise,
-    btcCoreAPI: BTCCoreAPI,
+    electrsAPI: ElectrsAPI,
     bitcoinCoreClient: BitcoinCoreClient,
-    keyring: Keyring,
-    amount: string,
-    requesterName: string,
-    vaultName: string,
-    autoExecute: boolean,
-    triggerRefund: boolean
+    issuingAccount: KeyringPair,
+    amount: Big,
+    vaultAddress?: string,
+    autoExecute = true,
+    triggerRefund = false
 ): Promise<IssueResult> {
     const treasuryAPI = new DefaultTreasuryAPI(api);
-    const issueAPI = new DefaultIssueAPI(api, bitcoin.networks.regtest, btcCoreAPI);
+    const issueAPI = new DefaultIssueAPI(api, bitcoin.networks.regtest, electrsAPI);
     const collateralAPI = new DefaultCollateralAPI(api);
 
-    const requester = keyring.addFromUri("//" + requesterName);
-    issueAPI.setAccount(requester);
-    const requesterAccountId = api.createType("AccountId", requester.address);
+    issueAPI.setAccount(issuingAccount);
```

Comment:
Not sure about this API, I think the signer should only be set on init.

Reply:
It depends. If the user selects a different account from the UI, we shouldn't have to re-initialize all the APIs, but rather just call `setAccount`. Having this exposed also helps in the integration tests

Reply:
Could the caller not call `setAccount` prior though?

Reply:
> we shouldn't have to re-initialize all the APIs

I think the overhead is fairly minimal compared to what we would need to read from stats

Reply:
My primary gripe is that this approach feels inconsistent if we have different signers on each component.

Reply:
This could be solved by passing an `account` reference (kind of like a singleton) to the TransactionAPI constructor (the parent class of the other APIs). In this setup, when `issueAPI.setAccount(issuingAccount);` is called, the account reference would change for all other APIs. What do you think?

Reply:
This would however count on the fact that the API objects are instantiated with the same account references - which would be easy to do in `polkabtc-api.ts` https://github.com/interlay/polkabtc-js/blob/master/src/polkabtc-api.ts#L76